r1cs/gadgets/crypto/src/crh/poseidon/mod.rs

```diff
@@ -170,7 +170,10 @@ impl<
     {
 
         // Check that the length of the state vector is t
-        assert_eq!(state.len(), P::T);
+        // TODO: check error message
```

Comment:
Let's keep the assertion as it's an internal function. Plus, it is enforced by construction